GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java

```diff
@@ -143,19 +146,34 @@ public SurveyInstance save(SurveyInstance si, DeviceFiles deviceFile) {
             Device d = deviceDao.getDevice(deviceFile.getAndroidId(),
                     deviceFile.getImei(), deviceFile.getPhoneNumber());
             String deviceId = d == null ? "null" : String.valueOf(d.getKey().getId());
+            ObjectMapper objectMapper = new ObjectMapper();
 
             for (QuestionAnswerStore qas : images) {
-                String filename = qas.getValue().substring(
-                        qas.getValue().lastIndexOf("/") + 1);
-
-                Queue queue = QueueFactory.getQueue("background-processing");
-                TaskOptions to = TaskOptions.Builder
-                        .withUrl("/app_worker/imagecheck")
-                        .param(ImageCheckRequest.FILENAME_PARAM, filename)
-                        .param(ImageCheckRequest.DEVICE_ID_PARAM, deviceId)
-                        .param(ImageCheckRequest.QAS_ID_PARAM, String.valueOf(qas.getKey().getId()))
-                        .param(ImageCheckRequest.ATTEMPT_PARAM, "1");
-                queue.add(to);
+                String value = qas.getValue();
+                String filename = null;            
+                if (value.startsWith("{")) { //JSON
+                    try {
+                        Map<String,String>imgMap = objectMapper.readValue(value,
```

Comment:
the object mapper is a memory sink so I was thinking that in this case, given that we dont need the whole JSON object but just the filename, could we use regex parsing instead?